### PR TITLE
check for missing template/sdk tags

### PR DESCRIPTION
**Pull request by rajatjindal**

### CODE REVIEW

File: .github/gh-checktags.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+set -e
+
+which gh &> /dev/null || {
+  echo "This script requires the Github CLI 'gh'"
+  exit 1
+}
+
+tags=$(gh release list --repo fermyon/spin --exclude-pre-releases --exclude-drafts | grep -v TITLE | awk '{print $1}')
+
+failed=0
```
Comment (rylev) [inline .github/gh-checktags.sh:16]: nit: can we just call this `exit` or `exit_code`?

File: .github/gh-checktags.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  fi
+
+  # check golang sdk tag
+  if [[ -z "$(git tag -l sdk/go/$tag)" ]]; then
+    echo "tag sdk/go/$tag does not exist"
+    failed=1
+  fi
+
+  # remove trailing .\d+ from the version
+  tag_1=`echo $tag | sed 's/\.[0-9]*$//g'`
```
Comment (rylev) [inline .github/gh-checktags.sh:33]: Nit:

```suggestion
  major_minor=`echo $tag | sed 's/\.[0-9]*$//g'`
```

**Review by rylev — COMMENTED** [review]
Looks good. I'm not sure a cron based approach is the ideal approach, but I can't think of a better strategy.

File: .github/gh-checktags.sh (hunk 2 above)
Comment (rajatjindal) [reply .github/gh-checktags.sh:33]: done 👍

File: .github/gh-checktags.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+tags=$(gh release list --repo fermyon/spin --exclude-pre-releases --exclude-drafts | grep -v TITLE | awk '{print $1}')
+
+failed=0
+for tag in $tags; do
+  if [[ $tag != v1* ]]; then
+    continue
+  fi
+
+  if [[ $tag == *rc* ]]; then
```
[the remark this answers is not in the document]
Comment (rajatjindal) [reply .github/gh-checktags.sh:22]: done 👍

File: .github/gh-checktags.sh (hunk 1 above)
Comment (rajatjindal) [reply .github/gh-checktags.sh:16]: done 👍

**Review by rylev — APPROVED** [review]

**Comment by rajatjindal** [comment]
the tests failures seems unrelated to this PR. so we can merge it, if it looks ok otherwise.

**Review by itowlson — APPROVED** [review]
As written, this will need updating when we move to v2.  But who know whether we will reassess how we do versioning at that point anyway, so it's probably safest to keep this v1-only for now.  Thanks!

**Comment by itowlson** [comment]
@rylev Re the cron approach: I get where you're coming from.  The workflow also has a `workflow_dispatch` trigger, so we could add "run the sanity check action" to the end of the release process.  And the cron at least gives us the "ambulance at the bottom of the cliff" if something goes amiss.

@rajatjindal Yes, the test failures are because of the known template/SDK mismatch issue.

**Pull request merged**